### ISSUE TITLE
Include country's postcode/zip format

### ISSUE THREAD

**Issue by olliebennett** [hexorx/countries#366]
I'd like a new field for each country to store the valid postal code / zip code syntax as a regular expression.

This would complement the `postal_code` (boolean) flag in allowing us to validate a country's postcode, for example.

The regex for `GB` postcodes is a monster but all other countries' postcodes seem quite simple.

A good data source is [CLDR](http://cldr.unicode.org/index/downloads) (via [StackOverflow](http://stackoverflow.com/questions/578406/what-is-the-ultimate-postal-code-and-zip-regex/7185241#7185241)).

For `US` (for example) the value would be `postal_code_format: "\d{5}([ \-]\d{4})?"`.

Would this be out of scope or should I provide a PR?

**Comment by rposborne** [comment]
This is absolutely in scope, massive bonus points of getting the data cleanly from CLDR.

**Comment by olliebennett** [comment]
Great - I'll sort out a PR when I get a chance.

On further investigation, it seems CLDR may not be the best source of info - according to [these release notes](http://cldr.unicode.org/index/downloads/cldr-27), postcode data was removed because it ...
> ... was not being maintained and there are other sources of such information that are kept up to date.

I've found references (such as [here](http://unicode.org/cldr/trac/ticket/7612)) to these "other sources", which seems to be:

http://i18napis.appspot.com/address/data/GB

Documentation about this Google-supported API is [here](https://github.com/googlei18n/libaddressinput/wiki/AddressValidationMetadata), and it seems to be functional and updated, although I can't find the raw dataset to scrape. Perhaps I'll have to process the responses from that API. Hopefully the data is sufficiently structured and simple - we'll have to keep postcode validation at the country level for now, as subregion-level data seems to be incomplete, at least through this API.
